components/feed/FilterSavedFeedButton.tsx

```diff
@@ -0,0 +1,43 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useState, useTransition } from "react";
+
+const FilterSavedFeedButton = ({
+  filterActive,
+}: {
+  filterActive?: boolean;
+}) => {
+  const router = useRouter();
+  const [isPending, startTransition] = useTransition();
+  const [isFetching, setIsFetching] = useState(false);
+
+  const isMutating = isPending || isFetching;
+
+  const handleClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setIsFetching(true);
+
+    setIsFetching(false);
+    startTransition(() => {
+      router.replace(`${filterActive ? "/" : "/saved"}`);
+    });
+  };
+
+  return (
+    <div className="flex flex-row">
```

Comment:
<img width="202" alt="image" src="https://user-images.githubusercontent.com/7117993/227063425-6d387c2a-0011-4d7f-b0d1-13a2dd578d0c.png">

youre missing something like `gap-4` and probably `items-center` to make it align nicely